spec/schemas/register_file_schema.json

```diff
@@ -0,0 +1,130 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+
+  "title": "Register File Schema",
+  "description": "Schema for describing a register file",
+
+  "$defs": {
+    "register_entry": {
+      "type": "object",
+      "required": ["name"],
+      "additionalProperties": false,
+      "properties": {
+        "name": {
+          "$ref": "schema_defs.json#/$defs/register_name"
+        },
+        "abi_mnemonics": {
+          "type": "array",
+          "items": {
+            "$ref": "schema_defs.json#/$defs/register_alias"
+          },
+          "minItems": 1,
+          "uniqueItems": true,
+          "description": "ABI mnemonic names for the register"
+        },
+        "description": {
+          "$ref": "schema_defs.json#/$defs/spec_text"
+        },
+        "when": {
+          "$ref": "schema_defs.json#/$defs/requires_entry"
+        },
+        "sw_read()": {
+          "type": "string",
+          "description": "Function that returns the value of the register when read by software. Use this to define special behavior for registers (e.g., x0 always reads as zero)."
```

Comment:
In a recent ad-hoc review in a Unified DB SIG meeting, we preferred that these `sw_read()` and `sw_write()` methods be renamed to something akin to "architectural read/write", so maybe `arch_read()` and `arch_write()`. This distinguishes configuration-dependent behavior from architecturally defined behavior. Please update the description to match.

The implementations can be kept the same, which is basically all for the case of `x0` to always be write-any-read-0.